### DISCUSSION TITLE
RabbitMQ should provide a method for blocking brute force attacks

### DISCUSSION THREAD

**Discussion by leaner1000** [rabbitmq/rabbitmq-server#4729] [Other]
RabbitMQ should provide a method for blocking brute force attacks
For security purposes, the user should be locked out after multiple incorrect password attempts.

**Comment by michaelklishin** [comment]
This is really light on details and does not live up to our basic problem definition quality standards. It's not specific, not actionable without lots of clarifications, and may or may not cover a core team consensus.

What exactly does "locked out" mean? For how long? How many incorrect attempts is too many? Over
what period of time? Why?
Why cannot existing tools help solve this? (e.g. TCP proxies have connection rate limiting with quite a few features these days).

How does a blocked user get unblocked?

These are the kind of questions that have to be clarified if this were to become an issue someone
from the core team would pick up.

At the moment, there is a metric that exposes the rate of failing authentication attempts and IIRC it includes some client connection details. So that helps with monitoring of such events.
